src/Analyzers/Core/Analyzers/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.cs

```diff
@@ -217,6 +217,7 @@ private bool IsUnusedParameterCandidate(IParameterSymbol parameter)
                     method.IsVirtual ||
                     method.IsOverride ||
                     method.PartialImplementationPart != null ||
+                    method.PartialDefinitionPart != null ||
```

Comment:
I don't know, but I feel the check is *too* strong.

Reply:
Given that
- a partial method definition might be generated by some tool, with a parameter list as part of its signature
- the corresponding implementation must have the same signature

I believe it's pointless to run the IDE0060 rule on the implementation (i.e. when `method.PartialDefinitionPart != null` ).

After all, if any of the parameters are unused by the implementation, what can the developer do? They probably have no control over the autogenerated definition and the implementation's signature must match the definition's...

Reply:
I think we should specifically check for generated code. This check can probably be easier when https://github.com/dotnet/roslyn/pull/63447/ is merged

Reply:
I didn't know about this upcoming `IsGeneratedCode` flag. You're right, waiting for it sounds like a good idea.

Reply:
I am not sure if `IsGeneratedCode` check helps here. The callback would only be made for non-generated code for IDE0060 analyzer, and you can't check whether or not the other partial symbol is generated or not from this callback.